### PR TITLE
Forgot password page

### LINKED ISSUE #149: Create forgot password page and component
Page should look like https://petitions.moveon.org/login/forgot_password.html and submit to API, but can be mock API for testing, since response is entirely server-side. After submit, should look like https://petitions.moveon.org/login/forgot_password_thanks.html

### PR DESCRIPTION
**Pull request by jparkrr**
closes #149
api issue: https://github.com/MoveOnOrg/mop/issues/150

Creates http://localhost:8080/#/login/forgot_password.html

For simplicity I have left out:
- All server errors (email is validated locally though)
- Thank you page doesn't have its own route or component
- Redux is not involved

I took a lot of the validation stuff from user registration. I'm not sure if it's worth extracting? https://github.com/MoveOnOrg/mop-frontend/blob/master/src/components/register-form.js

Open to changing anything!

### CODE REVIEW

File: src/actions/accountActions.js [hunk 1]
```diff
@@ -43,6 +43,17 @@ export function register(fields) {
   }
 }
 
+export function forgotPassword(email) {
+  // We don't care much about the response in this case
+  return fetch(`${Config.API_URI}/users/forgotPassword.json`, {
```
Comment (sreynen) [inline src/actions/accountActions.js]: Our API multi-word endpoints all have dashes between words, so let's do that here, e.g. `forgot-password.json`

File: src/components/forgot-password-form.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  errorList() {
+    return this.state.presubmitErrors.map((error, idx) => <li key={idx}>{error.message}</li>)
+  }
+
+  renderThanks() {
+    return (
+      <div>
+        An email has been sent containing a link you can use to reset your
+        MoveOn Petitions password.{' '}
+        <b>Thank you for using MoveOn's petition website!</b>
```
Comment (sreynen) [inline src/components/forgot-password-form.js]: Straight apostrophes mess up formatting on both GitHub and some of our editors, and curly apostrophes look a little better, so let's use `’` here.

File: src/components/forgot-password-form.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        >
+          <div className='control-group'>
+            <label className='control-label' htmlFor='inputEmail'>
+              Email:
+            </label>
+            <div className='controls'>
+              <input
+                ref={email => (this.email = email)}
+                name='email'
+                className='percent-80 validation_error'
```
Comment (sreynen) [inline src/components/forgot-password-form.js]: Are these classes doing something now that I'm missing, or is this for some future use?

Comment (jparkrr) [reply src/components/forgot-password-form.js]: Probably not. I just took them from the live site. I'll remove them if possible

Comment (jparkrr) [reply src/components/forgot-password-form.js]: looks like `percent-80` makes gives the input an 80% width, which makes it a bit more balanced IMO. I removed the others and the id. Thanks!

**Comment by jparkrr** [comment]
@sreynen thanks for the review! I've updated it

**Comment by jparkrr** [comment]
@sreynen I added tests here too

**Comment by sreynen** [comment]
Tests look good to me.

**Pull request merged**